src/main/scala/ai/privado/languageEngine/java/passes/config/JavaPropertyLinkerPass.scala

```diff
@@ -0,0 +1,111 @@
+/*
+ * This file is part of Privado OSS.
+ *
+ * Privado is an open source static code analysis tool to discover data flows in the code.
+ * Copyright (C) 2022 Privado, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * For more information, contact support@privado.ai
+ *
+ */
+
+package ai.privado.languageEngine.java.passes.config
+
+import io.shiftleft.codepropertygraph.generated.nodes.{
+  AnnotationParameterAssign,
+  JavaProperty,
+  Literal,
+  Member,
+  MethodParameterIn
+}
+import io.shiftleft.codepropertygraph.generated.{Cpg, EdgeTypes}
+import io.shiftleft.semanticcpg.language._
+import overflowdb.BatchedUpdate
+import overflowdb.traversal._
+import ai.privado.languageEngine.java.language.NodeStarters
+import ai.privado.tagger.PrivadoParallelCpgPass
+import io.shiftleft.passes.ForkJoinParallelCpgPass
+
+/** This pass creates a graph layer for Java `.properties` files.
+  */
+class JavaPropertyLinkerPass(cpg: Cpg) extends PrivadoParallelCpgPass[JavaProperty](cpg) {
+  override def generateParts(): Array[_ <: AnyRef] =
+    cpg.property.l.toArray.filter(pair => pair.name.nonEmpty && pair.value.nonEmpty)
```

Comment:
First filter on and then convert toArray, as here you are creating array 2 times